src/Tools/dotnet-monitor/CollectionRules/Triggers/ICollectionRuleTrigger.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers
+{
+    /// <summary>
+    /// Interface implemented by collection rule triggers.
+    /// </summary>
+    internal interface ICollectionRuleTrigger
+    {
+        /// <summary>
+        /// Waits for the trigger condition to be satisfied.
+        /// </summary>
+        Task WaitForSatisfiedConditionAsync(CancellationToken cancellationToken);
```

Comment:
So are you expecting that in the WaitForSatisfiedConditionAsync would effectively be a listener for the callback from ITraceEventTrigger (at least in many trigger cases)?

Reply:
That's the initial intent, however this interface is like to change. Just wanted something to start while I work out the initialization of the collection rule pipeline/execution.